aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -1133,6 +1142,24 @@ module aptos_framework::stake {
         assert!(exists<StakePool>(pool_address), error::invalid_argument(ESTAKE_POOL_DOES_NOT_EXIST));
     }
 
+    fun validate_validator_and_fullnode_addresses(network_addresses: vector<u8>, fullnode_addresses: vector<u8>) {
+        // Network addresses are optional but need to be valid if present.
+        if (vector::length(&network_addresses) > 0) {
+            assert!(
+                validate_network_addresses(network_addresses),
+                error::invalid_argument(EINVALID_VALIDATOR_ADDRESSES),
+            );
+        };
```

Comment:
I'm not sure if we wanted this to be required, normally, this is required, but full node is not.

Reply:
Updated